File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.py
def solution(board, moves):
    n = len(board)
    stack = []
    count = 0
    for m in moves:
        for i in range(n):
            if board[i][m-1] != 0:
                v = board[i][m-1]
                if stack and stack[-1]==v:
                    stack.pop()
                    count += 1
                else:
                    stack.append(v)
                board[i][m-1] = 0
                break
    return count*2

File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.java
import java.util.*;

class Solution {
    public int solution(int[][] board, int[] moves) {
        int answer= 0;
        Deque<Integer> deque = new ArrayDeque<>();
        int h = board[0].length;
        
        for(int move : moves) {
            for(int i=0; i<h; i++) {
                if(board[i][move-1] == 0) continue;
                
                int d = board[i][move-1];
                board[i][move-1] = 0;
                if(!deque.isEmpty() && deque.peek() == d){
                    deque.pop();
                    answer += 2;
                    break;
                }
                deque.push(d);
                break;
            }
        }
        
        return answer;
    }
}